formats/gltf/extensions.go

```diff
@@ -452,3 +460,37 @@ func (pd PolyformDispersion) ToExtensionData(w *Writer) map[string]any {
 	metadata["dispersion"] = pd.Dispersion
 	return metadata
 }
+
+// KHR_texture_transform ===================================================
+// https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_texture_transform
+
+// PolyformTextureTransform is a glTF extension that defines texture transformations.
+type PolyformTextureTransform struct {
+	Offset   *[2]float64 // The offset of the UV coordinate origin as a factor of the texture dimensions.
```

Comment:
Erm, it would make perfect sense to use `vector2.Float64` here, and I would go for it (more type safety is always better), but there is an issue - it will not serialise correctly in JSON.

You have this:
```
func (v Vector[T]) MarshalJSON() ([]byte, error) {
	return json.Marshal(&struct {
		X float64 `json:"x"`
		Y float64 `json:"y"`
	}{
		X: float64(v.x),
		Y: float64(v.y),
	})
}
```

So it will serialise as an object, while `Offset` and `Scale` require an array, something like `[1.0, 1.0]`.

I'm not sure where else `vector2` is used for JSON values, but it's incompatible and will have to be broken to make it work here. 

So I think we'll have to leave it as is, even though I like the idea.

Reply:
Wait, I'm an idiot, standby :D

Reply:
Done.